core/src/consensus/tendermint/engine.rs

```diff
@@ -173,6 +173,13 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
         if total_fee != min_fee {
             self.machine.add_balance(block, &author, total_fee - min_fee)?
         }
+
+        let current_metadata = block.state().metadata()?.unwrap();
+        let current_seq = current_metadata.seq();
+        let parent_common_seq = self.machine.metadata_seq(Some(block_number - 1));
+        if parent_common_seq != current_seq {
+            self.machine.insert_common_params(block_number, current_seq, current_metadata.params().clone().unwrap());
+        }
```

Comment:
What happen the block is not confirmed?
A proposal block always calls the `on_close_block`, but the proposal block may be not confirmed.

Reply:
Yes, you are right.
Do we have an event that is occurred when the block is confirmed?

Reply:
Yes, If we have the event, it will be perfect.

Reply:
I found that `ChainNotify`'s `new_blocks` is called when the best block is changed.
The new best block's hash is contained in the `enacted` parameter.

Reply:
I don't need this event anymore because the states store it.